sources/packages/backend/apps/api/src/route-controllers/supporting-user/_tests_/e2e/supporting-user.students.controller.submitSupportingUserDetails.e2e-spec.ts

```diff
@@ -0,0 +1,354 @@
+import { HttpStatus, INestApplication } from "@nestjs/common";
+import { TestingModule } from "@nestjs/testing";
+import * as request from "supertest";
+import * as faker from "faker";
+import {
+  BEARER_AUTH_TYPE,
+  createTestingAppModule,
+  FakeStudentUsersTypes,
+  getStudentToken,
+  mockDryRunSubmission,
+  mockJWTUserInfo,
+  resetMockJWTUserInfo,
+} from "../../../../testHelpers";
+import {
+  createE2EDataSources,
+  createFakeSupportingUser,
+  E2EDataSources,
+  saveFakeApplication,
+} from "@sims/test-utils";
+import {
+  Application,
+  ApplicationStatus,
+  DynamicFormConfiguration,
+  DynamicFormType,
+  SupportingUser,
+  SupportingUserType,
+} from "@sims/sims-db";
+import { ReportedSupportingUserAPIInDTO } from "../../../../route-controllers";
+import { AppStudentsModule } from "../../../../app.students.module";
+import { ZeebeGrpcClient } from "@camunda8/sdk/dist/zeebe";
+
+describe("SupportingUserStudentsController(e2e)-submitSupportingUserDetails", () => {
+  let app: INestApplication;
+  let appModule: TestingModule;
+  let db: E2EDataSources;
+  let recentPYParentForm: DynamicFormConfiguration;
+  let zeebeClient: ZeebeGrpcClient;
+
+  beforeAll(async () => {
+    const { nestApplication, module, dataSource } =
+      await createTestingAppModule();
+    app = nestApplication;
+    appModule = module;
+    db = createE2EDataSources(dataSource);
+    zeebeClient = app.get(ZeebeGrpcClient);
+    recentPYParentForm = await db.dynamicFormConfiguration.findOne({
```

Comment:
We have been using this approach in few E2E tests that it should be moved to a helper/factory 😉 
Not a blocker, I will do the next opportunity I have.